server/service/src/item/item_variant/upsert/generate.rs

```diff
@@ -0,0 +1,118 @@
+use chrono::Utc;
+use repository::{
+    item_variant::{item_variant::ItemVariant, item_variant_row::ItemVariantRow},
+    ActivityLogType, RepositoryError,
+};
+
+use crate::{activity_log::activity_log_entry, service_provider::ServiceContext};
+
+use super::UpsertItemVariantWithPackaging;
+
+pub fn generate(
+    user_id: &str,
+    UpsertItemVariantWithPackaging {
+        id,
+        name,
+        item_id,
+        cold_storage_type_id,
+        manufacturer_id,
+        packaging_variants: _, // Mapped separately
+        doses_per_unit,
+        vvm_type,
+    }: UpsertItemVariantWithPackaging,
+) -> ItemVariantRow {
+    ItemVariantRow {
+        id,
+        name,
+        item_link_id: item_id,
+        cold_storage_type_id: cold_storage_type_id.map(|l| l.value).unwrap_or_default(),
+        manufacturer_link_id: manufacturer_id
+            .map(|manufacturer_id| manufacturer_id.value)
+            .unwrap_or_default(),
+        deleted_datetime: None,
+        doses_per_unit,
+        vvm_type: vvm_type.map(|vvm_type| vvm_type.value).unwrap_or_default(),
+        created_datetime: Utc::now().naive_utc(),
+        created_by: Some(user_id.to_string()),
```

Comment:
oo also, this in an upsert, so would be overwriting this on every edit? probably we should use existing after creation?

Reply:
I think should fix this?